### PR TITLE
LeaveGroup lampham

### PR DESCRIPTION
**Pull request by LamPham123**
### **Overview** 
Leave group function allows user to leave the groups that they join.

### **Strategy**

**Backend (followed closely from createGroup):**

Create and define groupLeave constants  to keep track of states of our backend requests. (groupReducer.js, groupConstants.js, store.js)

Send request to the backend, passing groupId as parameter. (groupActions.js) 

Connect to frontend (groupRoutes) 

Function finds groupId and removes user from the members array in the database. If members array is empty (no members in group), delete the group. (groupControllers.js)

**Frontend:**

Create a leaveGroup function that takes a groupId parameter and confirms with the user if they want to leave the group. (GroupPage.js)

Dispatch leaveGroupAction, re-render the groupList on the left side and navigate back to "/myjournals" (GroupPage.js)

Used React icon, learned how to position the button and and cleaned up the css code (GroupPage.css)

### **Testing Evidence**
Screenshot 2024-06-25 at 1 19 19 AM

### CODE REVIEW

File: frontend/src/actions/groupActions.js [hunk 1]
```diff
@@ -11,9 +11,12 @@ import {
   GROUP_CODES_FAIL,
   GROUP_CODES_SUCCESS,
   GROUP_CODES_REQUEST,
+  GROUP_LEAVE_REQUEST,
+  GROUP_LEAVE_SUCCESS,
+  GROUP_LEAVE_FAIL,
 } from "../constants/groupConstants";
 
-const LOCAL = false;
+const LOCAL = true;
```
Comment (DavyHyun) [inline frontend/src/actions/groupActions.js]: change to false since it is being deployed

File: frontend/src/actions/groupActions.js [hunk 2]
```diff
@@ -130,6 +133,63 @@ export const addToGroup = (groupCode) => async (dispatch, getState) => {
...
+    }
+
+    if (!response.ok) {
+      const errorData = await response.json(); // Assuming server responds with JSON containing the error
+      throw new Error(errorData.message || "Something went wrong");
+    }
+
+    const data = await response.json();
+
+    console.log(data);
```
Comment (DavyHyun) [inline frontend/src/actions/groupActions.js]: remove console logs (in all cases)

File: frontend/src/screens/GroupPage/GroupPage.css [hunk 3]
```diff
@@ -7,3 +7,51 @@
...
+.header-content {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  position: relative;
+}
+
+.leave-group-icon {
+  position: absolute;
```
Comment (DavyHyun) [inline frontend/src/screens/GroupPage/GroupPage.css:42]: ngl i dont like position absolute did you check if this works for all screen sizes

File: frontend/src/screens/GroupPage/GroupPage.js [hunk 4]
```diff
@@ -273,36 +278,40 @@ const GroupPage = () => {
     }
   };
 
+  const leaveGroup = (id) => {
+    const group = groups.find(g => g._id === id);
+    const groupName = group.groupName;
+    const userResponse = window.confirm(
+      `Are you sure you want to leave group "${groupName}"?`
+    );
+    console.log(group)
```
Comment (DavyHyun) [inline frontend/src/screens/GroupPage/GroupPage.js]: console.log rm

File: frontend/src/screens/GroupPage/GroupPage.js [hunk 5]
```diff
@@ -273,36 +278,40 @@ const GroupPage = () => {
...
-        </div>
-        <SelectDateModal
-          show={showModal}
-          onHide={() => setShowModal(false)}
-          onApply={handleApply}
-          setShowModal={setShowModal} // Pass setShowModal as a prop
-          setShowAll={setShowAll} // Pass setShowAll as a prop
-        />
+      <div className="group-content">
+        <div className="group-header">
```
Comment (DavyHyun) [inline frontend/src/screens/GroupPage/GroupPage.js:300]: thanks for cleaning up css

**Comment by DavyHyun** [comment]
LGTM, fix minor nits i commented and can you post video (screen recording) of it working so I don't have to pull and test myself lol

**Review by DavyHyun — CHANGES_REQUESTED** [review]
thanks lam pham good work

**Comment by LamPham123** [comment]
**Testing Evidence**
Icon does work on screens of different sizes, and leave group does work, updates data base correctly as well 

https://github.com/DavyHyun/spiritual-journal/assets/63019320/1927f94e-e601-4dd4-8e5a-4fd5ccaaa726

**Pull request merged**